Show a list of installed plugins for Jetpack sites

koke opened the pull request:
First part of #7471:

- Feature flag: plugins only enabled on developer builds for now.
- Blog support flag: plugins only enabled for Jetpack sites via REST.
- UI: a basic plugin list with name, state, and a placeholder icon.
- Backend: Implemented remote and unit tests to load plugin list from the API.

![simulator screen shot 8 aug 2017 11 12 23](https://user-images.githubusercontent.com/8739/29064924-d3310e58-7c2a-11e7-89f4-052f624508b1.png)


To test:

- For a Jetpack site, go to Site > Plugins
- Verify that it list the currently installed plugins


Needs review: @elibud

elibud commented on WordPress/Classes/ViewRelated/Plugins/PluginListViewController.swift:
```diff
@@ -0,0 +1,79 @@
+class PluginListViewController: UITableViewController, ImmuTablePresenter {
+    fileprivate lazy var handler: ImmuTableViewHandler = {
+        return ImmuTableViewHandler(takeOver: self)
+    }()
```
Missing new line after this one?

elibud commented on WordPress/Classes/ViewRelated/Plugins/PluginListViewController.swift:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    let siteID: Int
+    let service: PluginServiceRemote
```
Personally I would add a new line after this line and I would move the convenience init to after the init below this comment, but that may be me just being particular about ordering things.

elibud commented on WordPressKit/WordPressKit/PluginState.swift:
```diff
@@ -0,0 +1,23 @@
+    var stateDescription: String {
+        switch (active, autoupdate) {
+        case (false, _):
+            return NSLocalizedString("Inactive", comment: "A plugin is not active on the site")
```
Instead of "A plugin"->"The plugin" ?

elibud replied:
We could also add a comment that it should be the same string as in WP.com

elibud commented:
One more question, in the description it says:

> a basic plugin list with name, version, and a placeholder icon.

But we are not currently showing the version but the state, right?

koke commented:
> we are not currently showing the version but the state

Correct, I changed that in the last commit after seeing the new mockups. I updated the screenshot but missed the description

koke commented:
Ready for another look

koke commented:
🎉 Thanks for the review!
